vispy/visuals/tests/test_rectangle.py

```diff
@@ -0,0 +1,135 @@
+        rectpolygon = visuals.Rectangle(pos=(0., 0.), height=20.,
+                                        width=20., radius=10.,
+                                        color='blue', border_color='red')
+        rectpolygon.transform = transforms.STTransform(scale=(2.0, 3.0),
+                                                       translate=(50, 50))
+        c.draw_visual(rectpolygon)
+        assert_image_equal("screenshot", 'visuals/rectpolygon7.png')
+
+        gloo.clear()
+        rectpolygon = visuals.Rectangle(pos=(0., 0.), height=60.,
+                                        width=60., radius=10.,
+                                        border_color='red')
+        rectpolygon.transform = transforms.STTransform(scale=(1.5, 0.5),
+                                                       translate=(50, 50))
+        c.draw_visual(rectpolygon)
+        assert_image_equal("screenshot", 'visuals/rectpolygon8.png', limit=0.7)
+
+        gloo.clear()
+        rectpolygon = visuals.Rectangle(pos=(0., 0.), height=60.,
+                                        width=60., radius=[25, 10, 0, 15],
+                                        color='blue', border_color='red')
+        rectpolygon.transform = transforms.STTransform(scale=(1.5, 0.5),
+                                                       translate=(50, 50))
+        c.draw_visual(rectpolygon)
+        assert_image_equal("screenshot", 'visuals/rectpolygon9.png')
+
+
+@requires_application()
+def test_reactive_draw():
+    """Test reactive RectPolygon attributes"""
+    with TestingCanvas() as c:
+        rectpolygon = visuals.Rectangle(pos=(50, 50, 0), height=40.,
+                                        width=80., color='red')
+        c.draw_visual(rectpolygon)
+
+        gloo.clear()
+        rectpolygon.radius = [20., 20, 0., 10.]
+        c.draw_visual(rectpolygon)
+        assert_image_equal("screenshot", 'visuals/reactive_rectpolygon1.png')
+
+        gloo.clear()
+        rectpolygon.pos = (60, 60, 0)
+        c.draw_visual(rectpolygon)
+        assert_image_equal("screenshot", 'visuals/reactive_rectpolygon2.png')
+
+        gloo.clear()
+        rectpolygon.color = 'blue'
+        c.draw_visual(rectpolygon)
+        assert_image_equal("screenshot", 'visuals/reactive_rectpolygon3.png')
+
+        gloo.clear()
+        rectpolygon.border_color = 'yellow'
+        c.draw_visual(rectpolygon)
+        assert_image_equal("screenshot", 'visuals/reactive_rectpolygon4.png')
+
+        gloo.clear()
+        rectpolygon.radius = 10.
+        c.draw_visual(rectpolygon)
+        assert_image_equal("screenshot", 'visuals/reactive_rectpolygon5.png')
+
+
+@requires_application()
+def test_attributes():
+    """Test if attribute checks are in place"""
+    with TestingCanvas():
+        rectpolygon = visuals.Rectangle(pos=(50, 50, 0), height=40.,
+                                        width=80., color='red')
+        assert_raises(ValueError, setattr, rectpolygon, 'height', 0.)
```

Comment:
Probably at some point, yeah. I'm willing to save that for a future wholesale `py.test` update later, though. (These are part of @mssurajkaiga's much earlier commits actually.)